sympy/combinatorics/perm_groups.py

```diff
@@ -5075,3 +5075,141 @@ def _stabilizer(degree, generators, alpha):
     return [_af_new(x) for x in stab_gens]
 
 PermGroup = PermutationGroup
+
+class SymmetricPermutationGroup(Basic):
+    """
+    The class defining the lazy form of SymmetricGroup
+
+    deg : int
+    """
+
+    def __new__(cls, deg):
+        deg = _sympify(deg)
+        obj = Basic.__new__(cls, deg)
+        obj._deg = deg
+        obj._order = None
+        return obj
+
+    def order(self):
+        """
+        Return the order of the permutation group.
+        """
+        if self._order is not None:
+            return self._order
+        n = self._deg
+        self._order = factorial(n)
+        return self._order
+
+    @property
+    def degree(self):
+        """
+        Return the degree of the permutation group.
+        """
+        return self._deg
+
+    @property
+    def identity(self):
+        '''
+        Return the identity element of the permutation group.
+
+        '''
+        return _af_new(list(range(self._deg)))
+
+class Coset(Basic):
+    """A left coset of a permutation group with respect to an element.
+
+    Parameters
+    ==========
+
+    g : Permutation
+
+    H : PermutationGroup
+
+    dir : "+" or "-", If not specified by default it will be "+"
+        here ``dir`` specified the type of coset "+" represent the
+        right coset and "-" represent the left coset.
+
+    G : PermutationGroup, optional
+        The group which contains *H* as its subgroup and *g* as its
+        element.
+
+        If not specified, it would automatically become a symmetric
+        group ``SymmetricPermutationGroup(g.size)`` and
+        ``SymmetricPermutationGroup(H.degree)`` if ``g.size`` and ``H.degree``
+        are matching.``SymmetricPermutationGroup`` is a lazy form of SymmetricGroup
+        used for representation purpose.
+    """
+
+    def __new__(cls, g, H, G=None, dir = "+"):
+        g = _sympify(g)
+        if not isinstance(g, Permutation):
+            raise NotImplementedError
+
+        H = _sympify(H)
+        if not isinstance(H, PermutationGroup):
+            raise NotImplementedError
+
+        if G is not None:
+            G = _sympify(G)
+            if not isinstance(G, PermutationGroup) and not isinstance(G, SymmetricPermutationGroup):
+                raise NotImplementedError
+            if isinstance(G, PermutationGroup):
+                if not H.is_subgroup(G):
+                    raise ValueError("{} must be a subgroup of {}.".format(H, G))
+                if g not in G:
+                    raise ValueError("{} must be an element of {}.".format(g, G))
+            else:
+                g_size = g.size
```

Comment:
Usually, these kinds of duplications are the reason why it's better to add trivial cases to is_subgroup for more general use, rather than trivial cases here.